### PR TITLE
Updating Nest/SmartThings/Insteon translators to use OpenT2TLogger

### PR DESCRIPTION
**Pull request by shalinijoshi19**
- Ran all CI-Checks - all pass locally.

### CODE REVIEW

**Comment by ChuckFerring** [comment]
:shipit:

File: org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js [hunk 1]
```diff
@@ -468,8 +474,8 @@ class Translator {
                 }
             })
             .catch(function (err) {
-                console.log("Request failed to: " + options.method + " - " + options.url);
-                console.log("Error            : " + err.statusCode + " - " + err.response.statusMessage);
+                this.ConsoleLogger.error(`Request failed to: ${options.method}- ${options.url}`);
```
Comment (supratiksen) [inline org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js]: nitpick missing space between ${options.method} and -

File: org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js [hunk 2]
```diff
@@ -177,8 +181,8 @@ class Translator {
                 return this._authTokens;
                 
             }).catch(function (err) {
-                console.log('Request failed to: ' + options.method + ' - ' + options.url);
-                console.log('Error            : ' + err.statusCode + ' - ' + err.response.statusMessage);
+                this.ConsoleLogger.error(`Request failed to: ${options.method}- ${options.url}`);
+                this.ConsoleLogger.error(`"Error            : ${err.statusCode} - ${err.response.statusMessage}`);
```
Comment (supratiksen) [inline org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js]: >" [](start = 42, length = 1)

Do we need the " after `

File: org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js [hunk 3]
```diff
@@ -177,8 +181,8 @@ class Translator {
                 return this._authTokens;
                 
             }).catch(function (err) {
-                console.log('Request failed to: ' + options.method + ' - ' + options.url);
-                console.log('Error            : ' + err.statusCode + ' - ' + err.response.statusMessage);
+                this.ConsoleLogger.error(`Request failed to: ${options.method}- ${options.url}`);
```
Comment (supratiksen) [inline org.opent2t.sample.hub.superpopular/com.insteon.hub/js/thingTranslator.js]: >- [](start = 78, length = 1)

nitpick missing space between ${options.method} and -

File: org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js [hunk 4]
```diff
@@ -249,15 +251,22 @@ class Translator {
...
+                        return errorMsg;
+                    }
                 } else {
                     return JSON.parse(body);
                 }                
             })
             .catch(function (err) {
-                console.log("Request failed to: " + options.method + " - " + options.url);
-                console.log("Error            : " + err);
+                this.ConsoleLogger.error(`Request failed to: ${options.method}- ${options.url}`);
```
Comment (supratiksen) [inline org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js]: >- [](start = 78, length = 1)

nitpick missing space between ${options.method} and -

File: org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js [hunk 5]
```diff
@@ -249,15 +251,22 @@ class Translator {
...
+                this.ConsoleLogger.error(`"Error            : ${err.statusCode} - ${err.response.statusMessage}`);
```
Comment (supratiksen) [inline org.opent2t.sample.hub.superpopular/com.smartthings.hub/js/thingTranslator.js]: >" [](start = 42, length = 1)

Do we need the " after `

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js [hunk 6]
```diff
@@ -248,7 +249,6 @@ function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
             break;
         case 'colourRGB':
             var HSVColor = RGBtoHSV(resourceSchema.rgbvalue);
-            console.log(HSVColor);
```
Comment (supratiksen) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: >console.log(HSVColor); [](start = 12, length = 22)

we no longer want to verbosely log this?

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js [hunk 7]
```diff
@@ -265,25 +265,24 @@ function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
             }
             break
         default:
-            // Error case
-            console.log("Invalid resourceId: " + resourceId)
```
Comment (supratiksen) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: >console.log("Invalid resourceId: " + resourceId) [](start = 12, length = 48)

don't want to log an error?

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js [hunk 8]
```diff
@@ -265,25 +265,24 @@ function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
...
-            throw new Error("Invalid resourceId");
+        throw new Error("Invalid resourceId");
```
Comment (supratiksen) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: >Error [](start = 18, length = 5)

opent2terror?

File: org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js [hunk 9]
```diff
@@ -23,13 +24,17 @@ function findResource(schema, di, resourceId) {
         return d.di === di; 
     }); 
     
-    if (!entity) throw new Error('NotFound');
+    if (!entity) {
+        throw new Error('NotFound');
```
Comment (supratiksen) [inline org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js]: >Error [](start = 18, length = 5)

opent2terror?

File: org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js [hunk 10]
```diff
@@ -23,13 +24,17 @@ function findResource(schema, di, resourceId) {
...
+    }
     
     var resource = entity.resources.find((r) => { 
         return r.id === resourceId;  
     }); 
 
-    if (!resource) throw new Error('NotFound'); 
+    if (!resource) {
+        throw new Error('NotFound'); 
+    }
```
Comment (supratiksen) [inline org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js]: opent2terror?

File: org.opent2t.sample.thermostat.superpopular/com.smartthings.thermostat/js/thingTranslator.js [hunk 11]
```diff
@@ -22,13 +23,17 @@ function findResource(schema, di, resourceId) {
         return d.di === di; 
     }); 
     
-    if (!entity) throw new Error('NotFound');
+    if (!entity) {
+        throw new Error('NotFound');
```
Comment (supratiksen) [inline org.opent2t.sample.thermostat.superpopular/com.smartthings.thermostat/js/thingTranslator.js]: >Error [](start = 18, length = 5)

opent2terror?

File: org.opent2t.sample.thermostat.superpopular/com.smartthings.thermostat/js/thingTranslator.js [hunk 12]
```diff
@@ -22,13 +23,17 @@ function findResource(schema, di, resourceId) {
...
+    }
     
     var resource = entity.resources.find((r) => { 
         return r.id === resourceId;  
     }); 
 
-    if (!resource) throw new Error('NotFound'); 
+    if (!resource) {
+        throw new Error('NotFound');
```
Comment (supratiksen) [inline org.opent2t.sample.thermostat.superpopular/com.smartthings.thermostat/js/thingTranslator.js]: >Error [](start = 18, length = 5)

opent2terror?

**Comment by supratiksen** [comment]
:shipit:

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js (hunk 8 above)
Comment (shalinijoshi19) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Being fixed by Dan in his PR. Will eventually need to merge his changes into mine once he checks in.

---
In reply to: [103830413](https://github.com/openT2T/translators/pull/188#discussion_r103830413) [](ancestors = 103830413)

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js (hunk 7 above)
Comment (shalinijoshi19) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: This was the only that was logging so for now made it consistent with the others; The errors today will get picked up and logged at the hubcontroller (higher) layer. Also need to refactor this a bit to allow function here to be able to call into the translator class scoped Logger variable

---
In reply to: [103830390](https://github.com/openT2T/translators/pull/188#discussion_r103830390) [](ancestors = 103830390)

File: org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js (hunk 9 above)
Comment (shalinijoshi19) [inline org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js]: Being fixed by Dan's changes

---
In reply to: [103830547](https://github.com/openT2T/translators/pull/188#discussion_r103830547) [](ancestors = 103830547)

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js (hunk 6 above)
Comment (shalinijoshi19) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Doesnt have access to the logger in this scope. Will need to figure out/refactor a bit so for now no.

---
In reply to: [103830317](https://github.com/openT2T/translators/pull/188#discussion_r103830317) [](ancestors = 103830317)

**Comment by danielrischar** [comment]
:shipit:

**Pull request merged**